### PR TITLE
Update database initialization

### PR DESCRIPTION
**Pull request by flapek**
- Refactor code
- Update database initialization
- update README files
- update .NET version 
- update packages version

### CODE REVIEW

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/AppInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            var dbContextTypes = AppDomain.CurrentDomain.GetAssemblies()
+               .SelectMany(x => x.GetTypes())
+               .Where(x => typeof(DbContext).IsAssignableFrom(x) && !x.IsInterface && x != typeof(DbContext));
+
+            using var scope = _serviceProvider.CreateScope();
+            foreach (var dbContextType in dbContextTypes)
+            {
+                if (scope.ServiceProvider.GetService(dbContextType) is not DbContext dbContext)
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Infrastructure/Services/AppInitializer.cs:27]: Doesn't if expression always be false?
I think that it's impossible to pass this condition because above you have `Where `which ensures that
`x => typeof(DbContext).IsAssignableFrom(x)`

Unless I missed sth.

File: TagsterWebAPI/src/Tagster.Infrastructure/EF/Configurations/ProfileConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Tagster.Domain.Entities;
+
+namespace Tagster.Infrastructure.EF.Configurations
+{
+    public class ProfileConfiguration : IEntityTypeConfiguration<Profile>
+    {
+        public void Configure(EntityTypeBuilder<Profile> builder)
+        {
+            builder.HasKey(t => t.Id);
```
Comment (kchowanski) [inline TagsterWebAPI/src/Tagster.Infrastructure/EF/Configurations/ProfileConfiguration.cs:11]: we followed EF Core conventions so those 2 lines are redundant - will work the same without them

**Review by kchowanski — COMMENTED** [review]
just a few not urgent comments/questions

File: TagsterWebAPI/src/Tagster.Infrastructure/Services/AppInitializer.cs (hunk 1 above)
Comment (kchowanski) [reply TagsterWebAPI/src/Tagster.Infrastructure/Services/AppInitializer.cs:27]: ok, I've got it :) there is a possibility to have class which is not registered in DI container, so skip this

**Review by kacperantoniak — APPROVED** [review]

File: TagsterWebAPI/src/Tagster.Infrastructure/Migrations/20211128112431_Init.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            migrationBuilder.DropTable(
+                name: "Users",
+                schema: "tagster");
+
+            migrationBuilder.DropTable(
+                name: "Profiles",
+                schema: "tagster");
+        }
+    }
+}
```
Comment (kacperantoniak) [inline TagsterWebAPI/src/Tagster.Infrastructure/Migrations/20211128112431_Init.cs:112]: Is this file required?

Comment (flapek) [reply TagsterWebAPI/src/Tagster.Infrastructure/Migrations/20211128112431_Init.cs:112]: yes, without this ef doesn't know how to initialize database
